src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEntryPointSymbol.cs

```diff
@@ -450,13 +429,12 @@ private sealed class ScriptEntryPoint : SynthesizedEntryPointSymbol
             //     var script = new Script();
             //     script.<Initialize>().GetAwaiter().GetResult();
             // }
-            internal override BoundBlock CreateBody()
+            internal override BoundBlock CreateBody(DiagnosticBag diagnostics)
             {
-                // CreateBody should only be called if no errors.
-                Debug.Assert((object)_getAwaiterMethod != null);
-                Debug.Assert((object)_getResultMethod != null);
-
                 var syntax = DummySyntax();
+                var compilation = _containingType.DeclaringCompilation;
+                var binder = compilation.GetBinder((CSharpSyntaxNode)compilation.SyntaxTrees[0].GetRoot());
```

Comment:
Why do we get the binder in this way?  It seems incorrect / arbitrary to grab the binder from the first syntax tree in the compilation.  Generally we grab it instead from a relative SyntaxNode.